soccer/gameplay/robocup-py.cpp

```diff
@@ -904,21 +1007,20 @@ BOOST_PYTHON_MODULE(robocup) {
         .def("run_pid_tuner", &OurRobot_run_pid_tuner)
         .def("end_pid_tuner", &OurRobot_end_pid_tuner)
         .def_readwrite("is_penalty_kicker", &OurRobot::isPenaltyKicker)
-        .def_readwrite("is_ball_placer", &OurRobot::isBallPlacer)
-        .def("is_facing", &OurRobot::isFacing);
+        .def_readwrite("is_ball_placer", &OurRobot::isBallPlacer);
 
     class_<OpponentRobot, OpponentRobot*, std::shared_ptr<OpponentRobot>,
            bases<Robot>>("OpponentRobot", init<Context*, int>());
 
-    class_<Ball, std::shared_ptr<Ball>>("Ball", init<>())
+    class_<BallState, BallState*>("Ball", init<>())
```

Comment:
Why the change from `shared_ptr` to raw ptr?

Reply:
We no longer have a reason to use `BallState` with `shared_ptr` values anywhere (it's a tiny little value type, there's no reason to avoid copies).